test/integ_tests/test_measure.py

```diff
@@ -0,0 +1,85 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import re
+
+import pytest
+from botocore.exceptions import ClientError
+
+from braket.aws.aws_device import AwsDevice
+from braket.circuits.circuit import Circuit
+from braket.devices import LocalSimulator
+
+DEVICE = LocalSimulator()
+SHOTS = 8000
+
+IONQ_ARN = "arn:aws:braket:us-east-1::device/qpu/ionq/Harmony"
+SIMULATOR_ARN = "arn:aws:braket:::device/quantum-simulator/amazon/sv1"
+OQC_ARN = "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy"
+
+
+@pytest.mark.parametrize("arn", [(IONQ_ARN), (SIMULATOR_ARN)])
+def test_unsupported_devices(arn):
+    device = AwsDevice(arn)
+    if device.status == "OFFLINE":
+        pytest.skip("Device offline")
+
+    circ = Circuit().h(0).cnot(0, 1).h(2).measure([0, 1])
+    error_string = re.escape(
+        "An error occurred (ValidationException) when calling the "
+        "CreateQuantumTask operation: Device requires all qubits in the program to be measured. "
+        "This may be caused by declaring non-contiguous qubits or measuring partial qubits"
```

Comment:
Nit: "measuring partial qubits" sounds like you're measuring parts of qubits. The "partial" of "partial measurement" refers to part of the qubit set, not parts of qubits. Maybe "or not measuring all qubits"? @rmshaffer feel free to weigh in here as a scientist, maybe I'm overthinking and this language is common

Reply:
I agree - "or not measuring all qubits" is clearer.
```suggestion
        "This may be caused by declaring non-contiguous qubits or not measuring all qubits."
```

Reply:
I also agree that 'not measuring all qubits' is clearer. 

This ValidationException is thrown on the service side so I don't think I can change the wording here in this PR.

Reply:
Oh right, good point